### PR TITLE
TF2 WeightNormalization

### PR DESCRIPTION
**Pull request by seanpmorgan**
Various changes to better fit TF2

### CODE REVIEW

File: tensorflow_addons/layers/python/wrappers.py [hunk 1]
```diff
@@ -13,17 +13,21 @@
...
+
 from tensorflow import name_scope
-from tensorflow.python.framework import ops
 from tensorflow.python.framework import tensor_shape
 from tensorflow.python.ops import array_ops
 from tensorflow.python.ops import nn_impl
+from tensorflow.python.ops import variables as tf_variables
+from tensorflow.python.ops.nn import moments
+from tensorflow.python.ops.math_ops import sqrt
+from tensorflow.python.ops.linalg_ops import norm
```
Comment (facaiy) [inline tensorflow_addons/layers/python/wrappers.py]: it's better in alphabetical order

Comment (facaiy) [inline tensorflow_addons/layers/python/wrappers.py]: it's better in alphabetical order

File: tensorflow_addons/layers/python/wrappers.py [hunk 2]
```diff
@@ -13,17 +13,21 @@
...
+
 from tensorflow.python.keras import initializers
-from tensorflow.python.eager import context
-from tensorflow.python.keras.engine.base_layer import Layer
-from tensorflow.python.keras.engine.base_layer import InputSpec
+from tensorflow.python.keras.engine.base_layer import Layer, InputSpec
```
Comment (facaiy) [inline tensorflow_addons/layers/python/wrappers.py]: I prefer to put them [on separate lines](https://www.python.org/dev/peps/pep-0008/#imports), or `from tensorflow.python.keras.engine import base_layer`. What do you think?

File: tensorflow_addons/layers/python/wrappers.py [hunk 3]
```diff
@@ -131,25 +128,27 @@ def build(self, input_shape):
                 trainable=True,
                 aggregation=tf_variables.VariableAggregation.MEAN)
 
-            with ops.control_dependencies([self.layer.g.assign(
-                    self._init_norm(self.layer.v))]):
-                self._compute_weights()
+            self.layer.g.assign(self._init_norm(self.layer.v))
```
Comment (facaiy) [inline tensorflow_addons/layers/python/wrappers.py]: We seems to drop the support for graph mode here, right? If yes, Is `@tf_test_util.run_all_in_graph_and_eager_modes` correct for test cases below?

File: tensorflow_addons/layers/python/wrappers_test.py [hunk 4]
```diff
@@ -53,31 +67,45 @@ def test_weightnorm_conv2d(self):
...
 
     @tf_test_util.run_all_in_graph_and_eager_modes
-    def test_weight_norm_tflayers(self):
+    def test_weightnorm_tflayers(self):
         images = random_ops.random_uniform((2, 4, 4, 3))
-        wn_wrapper = wrappers.WeightNormalization(layers.Conv2D(32, [2, 2]),
-                                         input_shape=(4, 4, 3))
+        wn_wrapper = wrappers.WeightNormalization(
+            keras.layers.Conv2D(32, [2, 2]),
+                                                  input_shape=(4, 4, 3))
```
Comment (facaiy) [inline tensorflow_addons/layers/python/wrappers_test.py]: Nit: indentation

File: tensorflow_addons/layers/python/wrappers_test.py [hunk 5]
```diff
@@ -53,31 +67,45 @@ def test_weightnorm_conv2d(self):
...
     @tf_test_util.run_all_in_graph_and_eager_modes
-    def test_weight_norm_nokernel(self):
+    def test_weightnorm_nokernel(self):
         with self.assertRaises(ValueError):
-            wrappers.WeightNormalization(layers.MaxPooling2D(2, 2)).build((2, 2))
+            wrappers.WeightNormalization(
+                keras.layers.MaxPooling2D(2, 2)).build((2, 2))
+
+    def test_weightnorm_keras(self):
+        input = np.random.random((10, 3, 4)).astype(np.float32)
```
Comment (facaiy) [inline tensorflow_addons/layers/python/wrappers_test.py]: Nit: builtin name, use input_data?

File: tensorflow_addons/layers/python/wrappers_test.py [hunk 6]
```diff
@@ -53,31 +67,45 @@ def test_weightnorm_conv2d(self):
...
+        outputs = testing_utils.layer_test(
+            wrappers.WeightNormalization,
+            kwargs={
+                'layer': keras.layers.Dense(2),
+                'input_shape': (3, 4)
+            },
+            input_data=input)
```
Comment (facaiy) [inline tensorflow_addons/layers/python/wrappers_test.py]: Can we use input_shape here?

File: tensorflow_addons/layers/python/wrappers_test.py (hunk 5 above)
Comment (facaiy) [inline tensorflow_addons/layers/python/wrappers_test.py]: Built-in name, use input_data instead?

File: tensorflow_addons/layers/python/wrappers.py (hunk 1 above)
Comment (seanpmorgan) [reply tensorflow_addons/layers/python/wrappers.py]: Agree.

File: tensorflow_addons/layers/python/wrappers.py (hunk 2 above)
Comment (seanpmorgan) [reply tensorflow_addons/layers/python/wrappers.py]: Agree, I pushed an update using base_layer

File: tensorflow_addons/layers/python/wrappers_test.py (hunk 4 above)
Comment (seanpmorgan) [reply tensorflow_addons/layers/python/wrappers_test.py]: Fixed thx

File: tensorflow_addons/layers/python/wrappers.py (hunk 3 above)
Comment (seanpmorgan) [reply tensorflow_addons/layers/python/wrappers.py]: Yeah, the test coverage on this layer is not sufficient but it was added early to help build out the directory structure. I will revisit and add additional tests, but for now this recommended initialization is not required to achieve the decoupled kernel. I added a TODO to revisit, but I'd like to merge prior to our first package release so some of this outdated structure is not in the release.

Comment (facaiy) [reply tensorflow_addons/layers/python/wrappers.py]: I agree.

**Comment by facaiy** [comment]
Is the py3 failures related?

**Pull request merged**